### PR TITLE
NETOBSERV-1076 Fix service change detection

### PR DESCRIPTION
**Pull request by jotak**
use DeepDerivative on labels & annotations only, rather than full ObjectMeta

### CODE REVIEW

File: pkg/helper/comparators.go [hunk 1]
```diff
@@ -75,7 +75,8 @@ func probeChanged(old, new *corev1.Probe) bool {
 }
 
 func ServiceChanged(old, new *corev1.Service, report *ChangeReport) bool {
-	return report.Check("Service meta changed", !equality.Semantic.DeepDerivative(new.ObjectMeta, old.ObjectMeta)) ||
+	return report.Check("Service annotations changed", !equality.Semantic.DeepDerivative(new.Annotations, old.Annotations)) ||
+		report.Check("Service labels changed", !equality.Semantic.DeepDerivative(new.Labels, old.Labels)) ||
```
Comment (stleerh) [inline pkg/helper/comparators.go:79]: What part of corev1.Service changed that triggered this to return true, or does it not matter because it's clear that only Annotations and Labels are relevant?  Why was this not happening before?

Comment (jotak) [reply pkg/helper/comparators.go:79]: Many parts of the `ObjectMeta` will differ when comparing an existing object with a newly built object; e.g. are the uuid, creation date, etc. It's safe to rely only on labels & annotations as we aren't because that's at the moment the only things in `ObjectMeta` that we partly derive from configuration*. You can see an example here of a service that we create: https://github.com/netobserv/network-observability-operator/blob/main/controllers/consoleplugin/consoleplugin_objects.go#L300-L322

*: with the exception of the namespace, but a change of namespace triggers a whole different reconciliation process.

Why it happens only now? I wasn't sure... but now I remember something: in https://github.com/netobserv/network-observability-operator/pull/326, during the refactoring I fixed something related to the `DidChange` / `IsInProgress` stuff: previously it wasn't fully working in every situation because these were [fields of a structure that happened to be sometimes copied as argument](https://github.com/netobserv/network-observability-operator/pull/326/files#diff-7340f1cb3173df7c142997d58f7230bfd7aa1564f65be4ca3f88abe87e7d6a0dL19-L23) - which means that the "didChange" field of the parent was never updated when the one from the copy was. I fixed it by [making didChange/isInProgress variables now only created and owned from the parent, and updated via lambdas](https://github.com/netobserv/network-observability-operator/pull/326/files#diff-b1b7336af11d201e740a99a59ac20aa4845637c2b80f3eda38d1b03861ed74c4R102-R104).

So, tl;dr:  it *was* happening before, but without the same impact on the "Updating" CR status.

Comment (stleerh) [reply pkg/helper/comparators.go:79]: Thanks for the detailed explanation.
